File: core/src/main/kotlin/com/sealkrach/politirush/core/GameConfig.kt
package com.sealkrach.politirush.core

/** Constantes d'équilibrage. Tout ce qui se règle "au feeling" est ici. */
object GameConfig {
    const val HERO_MAX_HP = 3
    const val START_FIREPOWER = 1
    const val MIN_FIREPOWER = 1
    const val MAX_FIREPOWER = 64

    /** Hauteur à laquelle le héros se tient (y). */
    const val HERO_Y = 0.08f
    /** Rayon de collision du héros (pour les pièces et bonus). */
    const val HERO_RADIUS = 0.06f
    /** Rayon de collision d'une cible. */
    const val ENEMY_RADIUS = 0.05f
    /** Rayon de collision d'un projectile. */
    const val PROJECTILE_RADIUS = 0.015f
    /** Vitesse verticale des projectiles (fraction de hauteur / s). */
    const val PROJECTILE_SPEED = 1.4f
    /** Nombre maximum de projectiles simultanés (garde-fou performance). */
    const val MAX_PROJECTILES = 300
    /** Largeur totale de l'éventail de tir, en fraction de largeur, pour firepower projectiles. */
    const val SPREAD_WIDTH = 0.35f

    /** Vitesse de défilement de la piste (portes, bonus, pièces). */
    const val SCROLL_SPEED = 0.35f
    /** Intervalle entre deux paires de portes, en secondes. */
    const val GATE_INTERVAL = 6f
    /** Intervalle entre deux bonus, en secondes. */
    const val BONUS_INTERVAL = 11f

    /** Durée d'une vague en secondes. */
    const val WAVE_DURATION = 20f
    /** Intervalle de base entre deux apparitions de cibles (diminue avec les vagues). */
    const val BASE_SPAWN_INTERVAL = 1.1f
    const val MIN_SPAWN_INTERVAL = 0.25f
    /** Une vague sur N se termine par un boss. */
    const val BOSS_EVERY_N_WAVES = 5

    /** Temps sans élimination avant que le combo retombe à zéro. */
    const val COMBO_WINDOW = 2.5f
    /** Multiplicateur de score par palier de combo (chaque tranche de 10). */
    const val COMBO_STEP = 10

    /** Multiplicateur de cadence pendant la rafale. */
    const val RAFALE_FACTOR = 3f
    const val RAFALE_DURATION = 4f
    const val UPPERCUT_RADIUS = 0.35f
    const val LAMES_DURATION = 5f
    const val GADGET_DURATION = 4f
    const val COMPETENCES_DURATION = 6f
}

File: Assets/PolitiRush/Core/GameConfig.cs
namespace PolitiRush.Core
{
    /// <summary>Constantes d'équilibrage. Miroir de GameConfig.kt.</summary>
    public static class GameConfig
    {
        public const int HeroMaxHp = 3;
        public const int StartFirepower = 1;
        public const int MinFirepower = 1;
        public const int MaxFirepower = 64;

        public const float HeroY = 0.08f;
        public const float HeroRadius = 0.06f;
        public const float EnemyRadius = 0.05f;
        public const float ProjectileRadius = 0.015f;
        public const float ProjectileSpeed = 1.4f;
        public const int MaxProjectiles = 300;
        public const float SpreadWidth = 0.35f;

        public const float ScrollSpeed = 0.35f;
        public const float GateInterval = 6f;
        public const float BonusInterval = 11f;

        public const float WaveDuration = 20f;
        public const float BaseSpawnInterval = 1.1f;
        public const float MinSpawnInterval = 0.25f;
        public const int BossEveryNWaves = 5;

        public const float ComboWindow = 2.5f;
        public const int ComboStep = 10;

        public const float RafaleFactor = 3f;
        public const float RafaleDuration = 4f;
        public const float UppercutRadius = 0.35f;
        public const float LamesDuration = 5f;
        public const float GadgetDuration = 4f;
        public const float CompetencesDuration = 6f;
    }
}
